test/Test/Tinfoil/Arbitrary.hs

```diff
@@ -95,3 +95,9 @@ instance Arbitrary InvocationCount where
 
 instance Arbitrary InvocationField where
   arbitrary = InvocationField <$> arbitrary <*> arbitrary
+
+instance Arbitrary FixedField where
+  arbitrary = FixedField <$> choose (0, 2^(32 :: Int) - 1)
```

Comment:
That `choose` thing is identical to `arbitrary`.

You're right that the arbitrary instance for `Int` is weird and doesn't give anything *near* a uniform distribution, bit the instance for `Word32` is more than fine. If it was by code I'd just change that to `arbitrary` and maybe add a comment that the type is `Word32` and that the `Arbitrary` instance for that type is known to be sane.